subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -20,13 +20,36 @@
 
 LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
 
+/* In order to transmit the maximum size we choose CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU
+ * as the buffer size. However, dir_list_object_encode function encodes one full
+ * object record at a time so the buffer must be a multiple of object record length.
+ */
+#define OTS_DIR_LIST_BUFFER_SIZE (DIR_LIST_OBJ_RECORD_MAX_SIZE * \
+	ceiling_fraction(CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU, DIR_LIST_OBJ_RECORD_MAX_SIZE))
```

Comment:
I think I finally understand it. Since we always want to encode a full object (instead of e.g. 2/3rds of an object followed by 1/3rd of an object), we need multiple of `DIR_LIST_OBJ_RECORD_MAX_SIZE` to ensure that's possible. 

That makes sense (even though it took me a while to understand) :)